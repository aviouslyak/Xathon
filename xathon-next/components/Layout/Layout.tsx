import React from "react";

const Layout: React.FC = ({ children }) => {
  return (
    <div
      className="w-screen h-screen bg-rose-400 text-slate-800"
      data-testid="page-wrapper"
    >
      <div className="flex w-full justify-center">
        <h1 className="text-6xl font-header border-2 border-rose-900 p-2 my-4 rounded-md shadow-md shadow-pink-900">
          Xathon
        </h1>
      </div>
      {children}
    </div>
  );
};
export default Layout;
