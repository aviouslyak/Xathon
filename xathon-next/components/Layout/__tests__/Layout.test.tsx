import { render, screen } from "@testing-library/react";
import Layout from "../Layout";

describe("<Layout />", () => {
  let component: HTMLElement;
  beforeEach(() => {
    render(
      <Layout>
        <p data-testid="test-child">Foo</p>
      </Layout>
    );
    component = screen.getByTestId("page-wrapper");
  });

  test("layout matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });

  test("layout renders children", () => {
    const child = screen.getByTestId("test-child");
    expect(child).toBeInTheDocument();
  });
});
