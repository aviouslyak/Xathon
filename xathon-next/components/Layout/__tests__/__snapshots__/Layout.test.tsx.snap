// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> layout matches snapshot 1`] = `
<div
  class="w-screen h-screen bg-rose-400 text-slate-800"
  data-testid="page-wrapper"
>
  <div
    class="flex w-full justify-center"
  >
    <h1
      class="text-6xl font-header border-2 border-rose-900 p-2 my-4 rounded-md shadow-md shadow-pink-900"
    >
      Xathon
    </h1>
  </div>
  <p
    data-testid="test-child"
  >
    Foo
  </p>
</div>
`;
